saleor/tests/e2e/orders/test_update_draft_order_metadata.py

```diff
@@ -0,0 +1,146 @@
+import pytest
+
+from .. import DEFAULT_ADDRESS
+from ..metadata.utils import (
+    delete_metadata,
+    delete_private_metadata,
+    update_metadata,
+    update_private_metadata,
+)
+from ..product.utils.preparing_product import prepare_product
+from ..shop.utils.preparing_shop import prepare_default_shop
+from ..utils import assign_permissions
+from .utils import (
+    draft_order_complete,
+    draft_order_create,
+    order_lines_create,
+    order_query,
+    order_update_shipping,
+)
+
+
+@pytest.mark.e2e
+def test_update_metadata_draft_order_CORE_0245(
+    e2e_staff_api_client,
+    shop_permissions,
+    permission_manage_product_types_and_attributes,
+    permission_manage_orders,
+):
+    permissions = [
+        *shop_permissions,
+        permission_manage_product_types_and_attributes,
+        permission_manage_orders,
+    ]
+    assign_permissions(e2e_staff_api_client, permissions)
+
+    product_price = 33.33
+
+    shop_data = prepare_default_shop(e2e_staff_api_client)
+    channel_id = shop_data["channel"]["id"]
+    warehouse_id = shop_data["warehouse"]["id"]
+    shipping_method_id = shop_data["shipping_method"]["id"]
+
+    (
+        _product_id,
+        product_variant_id,
+        product_variant_price,
+    ) = prepare_product(
+        e2e_staff_api_client, warehouse_id, channel_id, variant_price=product_price
+    )
+
+    # Step 1 - Create a draft order
+    input = {
+        "channelId": channel_id,
+        "userEmail": "customer@example.com",
+        "billingAddress": DEFAULT_ADDRESS,
+        "shippingAddress": DEFAULT_ADDRESS,
+    }
+    data = draft_order_create(e2e_staff_api_client, input)
+    order_id = data["order"]["id"]
+    assert data["order"]["billingAddress"] is not None
+    assert data["order"]["shippingAddress"] is not None
+    assert order_id is not None
+
+    # Step 2 - Add metadata to draft order
+    metadata_input = [
+        {"key": "metadataToBeUpdated", "value": "test value"},
+        {"key": "metadataToBeRemoved", "value": "A"},
+        {"key": "testEntityCode", "value": "A"},
```

Comment:
Those key and values are used multiple times, IMo it's worth creating variables for them